sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/LocalDocumentStorage.java

```diff
@@ -234,8 +230,30 @@ boolean deleteOnline(String table, String partition, String documentId) {
         }
     }
 
-    List<LocalDocument> getDocumentsByPartition(String table, String partition) {
-        return queryLocalStorage(table, PARTITION_COLUMN_NAME + " = ?", new String[]{partition});
+    <T> DocumentWrapper<T> createOrUpdateOffline(String table, String partition, String documentId, T document, Class<T> documentType, WriteOptions writeOptions) {
+        DocumentWrapper<T> cachedDocument = read(table, partition, documentId, documentType, null);
+        cachedDocument.setFromCache(true);
+        if (cachedDocument.getError() != null && cachedDocument.getError().getMessage().equals(FAILED_TO_READ_FROM_CACHE)) {
+            return cachedDocument;
+        }
+
+        /* The document cache has been expired, or the document did not exists, create it. */
+        DocumentWrapper<T> writeDocument =
+                new DocumentWrapper<>(
+                        document,
+                        partition,
+                        documentId,
+                        cachedDocument.getETag(),
+                        System.currentTimeMillis() / 1000L);
```

Comment:
Wait this is a breaking change if we store seconds instead of ms.
The previous code was doing the conversion in memory, it didn't reach storage.

Reply:
This code does not reach sqlite, my bad.